docs/src/catalyst_applications/jump_simulation_performance.md

```diff
@@ -0,0 +1,163 @@
+# [Advice for performant jump simulations](@id jump_simulation_performance)
+We have previously described how to perform *jump simulations* of *chemical reaction network* (CRN) models using e.g. Gillespie's algorithm. These simulations can, however, be highly computationally intensive. Fortunately, there are several ways to increase their performance (thus reducing runtime). Here, we describe various considerations for performant jump simulations. Furthermore, all jump simulations of Catalyst models are performed using the JumpProcesses.jl package, [which documentation](https://github.com/SciML/JumpProcesses.jl) provides a more extensive description of these.
+
+#### Brief (and optional) introduction to jump simulations
+Jump simulations are continuous time, discrete space, simulations (where the system's state, throughout the simulation, are the discrete copy numbers of each species). The system's state changes at discrete time points (in so-called jumps). For CRNs, these jumps correspond to the occurrence of individual reactions. Here, *stochastic chemical kinetics* describes how the reactions are distributed across time. Typically, the frequency of each reaction depends on its *propensity* (which in turn depends on its *rate* and *substrates*). These depend on the state of the system, and must thus be re-computed whenever the system's state changes. Hence, jump simulations' runtimes are heavily dependent on how frequently these propensities must be recomputed.
+
+Typically, propensities are recomputed only whenever a jump occurs. This means that jump simulations' runtimes, very roughly, scale with the number of jumps. Runtimes typically become prohibitively expensive for:
+- Simulations of large models (either with many different species or where some species occur in large copy numbers).
+- Simulations over long time spans.
+- Simulations that are performed a large number of times.
+
+More throughout description of these types of simulations can be found in the literature [^1].
+
+## Managing of solution saving
+By default, `solve` saves the value of the solution at the time of every jump. For simulations where a large number of jumps occur, this can cause memory to quickly fill up. Typically, for simulations with a large number of jumps, we want to [disable this feature](https://docs.sciml.ai/JumpProcesses/dev/tutorials/discrete_stochastic_example/#save_positions_docs) and instead set the save frequency manually. To exemplify this, let us consider a simple production/degradation model:
```

Comment:
I agree. ish. Basically, at some point I think it would make sense to split out model simulation in the same manner as we did inverse problems. When we do, I think it would be useful to have a page introduction simulations (more properly than in the introduction to Catalyst). It would be one tutorial, with one section each for ODE, SDE, and Jumps. I think this should go there, and then this tutorial can make references to that. Basically, this is not supposed to become the master tutorial for Jump simulations, but rather a continuation on the general introduction.

I agree this order of things might be not the best though. Do you think it would make sense to do it this way? In that case, I can make a PR to:
- Create a simulation section
- Create an introduction to simulations (could also cover some additional useful stuff, like difference between DIfferentialEquartions and OrdinaryDiffEq/StochasticDiffEq/JumpProcesses).

I think it would also make sense to move the part where we show the equations (e.g. explaining how higher order terms are handled) from the Introduction to Catalyst tutorial to this tutorial.

Next, this and the ODE performance tutorial could be rebased on that, and we might consider what to put in this tutorial (given what already is in the base one).